server/src/test/java/org/elasticsearch/common/util/concurrent/TimedRunnableTests.java

```diff
@@ -144,4 +136,25 @@ public void testTimedRunnableRethrowsRejectionWhenNotAbstractRunnable() {
         assertSame(exception, thrown);
     }
 
+    public void testTimedRunnableExecutesNestedOnAfterOnce() {
```

Comment:
Can you add a test for the onFailure double invocation too?

Reply:
Sure, added test in a89c0e7bfe2